refactor(Components): Create wrappers for Ant Design 5 direct exports

EnxDev opened the pull request:
<!---
Please write the PR title following the conventions at https://www.conventionalcommits.org/en/v1.0.0/
Example:
fix(dashboard): load charts correctly
-->

### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
In this PR, I have reorganized the components within src/components/index.tsx, placing each component into its own directory under src/components/{component}.

Additionally, all components now include their respective stories and test files for better documentation and testing coverage.

As part of the changes, I also updated all import statements across the project to reflect the new structure. 
This refactor improves the overall organization, making it more maintainable and scalable in the future.

Furthermore, this PR migrates the Upload component from Ant Design 4 to Ant Design 5.  (Related to [this](https://github.com/apache/superset/issues/32824#event-16978018769))

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->
- AutoComplete Story
![autocomplete](https://github.com/user-attachments/assets/6df67af1-8f3b-465d-98bf-17f028d25f48)
- Grid Story
![grid](https://github.com/user-attachments/assets/680ed172-4993-4f47-8d50-6f62fd531193)
- Layout Story
![layout](https://github.com/user-attachments/assets/1160f5fc-a21c-439a-a056-132d209dc88a)
- Tree Story
![tree](https://github.com/user-attachments/assets/2f9a9c88-80b9-457e-943f-aa4a4b045519)
- Upload Story
![upload](https://github.com/user-attachments/assets/9d2e9266-3b79-4785-9aa0-c9c9305fa5ec)

### TESTING INSTRUCTIONS
<!--- Required! What steps can be taken to manually verify the changes? -->
- All tests should pass

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

EnxDev committed:
refactor(AutoComplete): add wrapper, tests, and story for Ant Design 5 AutoComplete component

EnxDev committed:
refactor(Tree): add wrapper, tests, and story for Ant Design 5 Tree component

EnxDev committed:
refactor(Grid): add wrapper, tests, and story for Antd5 Col and Row components

EnxDev committed:
refactor(Layout): add wrapper, tests, and story for Ant Design 5 Layout component

EnxDev committed:
Merge branch master into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
Merge branch master into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
chore: add missing license headers

EnxDev committed:
feat(storybook): migrate stories to CSF3 for Tree, AutoComplete, Grid, and Layout

EnxDev committed:
refactor(RTL tests): improve tests for AutoComplete and Tree components

EnxDev committed:
chore: replace Col and Row imports with direct imports from Grid

msyavuz commented on superset-frontend/src/components/AutoComplete/index.tsx:
```diff
@@ -0,0 +1,25 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import AutoComplete, {
+  AutoCompleteProps as AntAutoCompleteProps,
+} from 'antd-v5/lib/auto-complete';
+
+export type { AntAutoCompleteProps };
```
This renaming seems unneccessary

msyavuz commented on superset-frontend/src/components/AutoComplete/AutoComplete.stories.tsx:
```diff
@@ -0,0 +1,208 @@
+      <AutoCompleteWithOptions {...args} />
+    </div>
+  ),
+};
```
I don't think this story is functional

EnxDev replied:
It should works

msyavuz commented on superset-frontend/src/components/Grid/index.tsx:
```diff
@@ -0,0 +1,29 @@
+
+export { Row, Col };
+
+const Grid = { useBreakpoint };
```
I think we use `Object.assign` syntax in multiple places. And useBreakpoint import here seems unneccessary? We can probably assign to `Grid.useBreakpoint`

EnxDev replied:
I would prefer not to assign a hook to the object; I want like to maintain the intended usage in the library with destructuring from Grid.

msyavuz commented on superset-frontend/src/components/Input/index.tsx:
```diff
@@ -17,10 +17,13 @@
 export const InputNumber = AntdInputNumber;
 
-export const { TextArea } = AntdInput;
+export const Input = Object.assign(Antd5Input, {
+  TextArea: Antd5Input.TextArea,
+  Search: Antd5Input.Search,
+  Password: Antd5Input.Password,
+  OTP: Antd5Input.OTP,
+});
```
This seems better!

msyavuz commented on superset-frontend/src/components/Grid/Grid.test.tsx:
```diff
@@ -0,0 +1,36 @@
+ * under the License.
+ */
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
```
is this neccessary?

EnxDev replied:
Yes, without it, the test would fail.

msyavuz commented on superset-frontend/src/components/Layout/Layout.stories.tsx:
```diff
@@ -0,0 +1,126 @@
+              }}
+            />
+            <Menu defaultSelectedKeys={['1']} mode="inline">
+              <Menu.Item key="1" icon={<MenuUnfoldOutlined />}>
```
Let's use the `options` prop on menu instead

msyavuz commented on superset-frontend/src/components/Layout/Layout.stories.tsx:
```diff
@@ -0,0 +1,126 @@
+    const breakpoint: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl' = 'md';
+    const headerVisible = true;
+    const footerVisible = true;
+    const [collapsed, setCollapsed] = useState(false); // Solo questa rimane con useState
```
Instead of this state we can use the args here as well to make it interactive. Also this story can only control hasSider

EnxDev replied:
We can map const inside the component, because the only prop available in the Layout component is hasSider. Different props are available for Layout.Sider

msyavuz commented on superset-frontend/src/components/Layout/index.tsx:
```diff
@@ -0,0 +1,23 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import Layout from 'antd-v5/lib/layout';
```
This should probably be `es` as well. And now that i think about it maybe this pr should also be based off of theming branch?

msyavuz commented on superset-frontend/src/components/Tree/index.tsx:
```diff
@@ -0,0 +1,25 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import Tree from 'antd-v5/lib/tree/Tree';
```
Same `lib` problem here

msyavuz commented on superset-frontend/src/components/Tree/index.tsx:
```diff
@@ -0,0 +1,25 @@
+ */
+import Tree from 'antd-v5/lib/tree/Tree';
+import type { TreeProps, DataNode as TreeDataNode } from 'antd-v5/es/tree';
+import DirectoryTree from 'antd-v5/lib/tree/DirectoryTree';
```
Instead of importing from lib we should get it from Tree object.

geido committed:
chore: Merge template_less

geido committed:
chore(Antd): Import from es

msyavuz commented on superset-frontend/src/components/AutoComplete/index.tsx:
```diff
@@ -0,0 +1,25 @@
+ * under the License.
+ */
+
+import AutoComplete from 'antd-v5';
```
```suggestion
import { AutoComplete } from 'antd-v5';
```

msyavuz commented on superset-frontend/src/components/Tree/index.tsx:
```diff
@@ -0,0 +1,25 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import Tree from 'antd-v5';
```
```suggestion
import { Tree } from 'antd-v5';
```

msyavuz commented on superset-frontend/src/components/Layout/index.tsx:
```diff
@@ -16,6 +16,8 @@
  * under the License.
  */
-import { Layout } from 'antd-v5';
+import Layout from 'antd-v5';
```
```suggestion
import { Layout } from 'antd-v5';
```

geido committed:
Update superset-frontend/src/components/AutoComplete/index.tsx

Co-authored-by: Mehmet Salih Yavuz <salih.yavuz@proton.me>

geido committed:
Update superset-frontend/src/components/Layout/index.tsx

Co-authored-by: Mehmet Salih Yavuz <salih.yavuz@proton.me>

geido committed:
Update superset-frontend/src/components/Tree/index.tsx

Co-authored-by: Mehmet Salih Yavuz <salih.yavuz@proton.me>

EnxDev commented on superset-frontend/src/components/Tree/index.tsx:
```diff
@@ -0,0 +1,25 @@
+
+export type { TreeProps };
+export { TreeDataNode, DirectoryTree };
+export default Tree;
```
Ok, we can use that approach, thanks

msyavuz replied:
Just keeping this unresolved as a reminder for named exports

EnxDev replied:
We were thinking of opening a separate PR to apply named exports to all components, @geido right?

geido replied:
Yes I will fix named exports everywhere in a separate PR

geido committed:
chore(Antd): Fix Grid import / export

msyavuz commented on superset-frontend/src/components/Tree/index.tsx:
```diff
@@ -0,0 +1,25 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Tree } from 'antd-v5';
+import type { TreeProps, DataNode as TreeDataNode } from 'antd-v5/es/tree';
+import DirectoryTree from 'antd-v5/es/tree/DirectoryTree';
+
+export type { TreeProps };
+export { TreeDataNode, DirectoryTree };
+export default Tree;
```
```suggestion
import { Tree } from 'antd-v5';

export type { TreeProps, DataNode as TreeDataNode } from 'antd-v5/es/tree';
export default Tree;
```

geido committed:
Update superset-frontend/src/components/Tree/index.tsx

Co-authored-by: Mehmet Salih Yavuz <salih.yavuz@proton.me>

geido committed:
chore(Antd): Fix Grid imports

EnxDev committed:
refactor(storybook): remove constants in favor of static values in layout stories

EnxDev committed:
refactor(tag): change import from Antd to Antd5

EnxDev committed:
refactor(Tag): upgrade Tag and TagList
- Create two separate wrappers, one for Tag and one for TagsList
- Update imports to reflect changes
- Migrate stories from CSF2 to CSF3 format

EnxDev committed:
feat(skeleton): add wrapper with tests and stories
- Create a wrapper for the Skeleton component
- Add unit tests using RTL
- Implement CSF3 stories for Skeleton

EnxDev committed:
feat(typography): add wrapper with tests and stories
- Create a wrapper for the Typography component
- Add unit tests using RTL
- Implement CSF3 stories for Typography

EnxDev committed:
feat(TreeSelect): add wrapper with tests and stories
- Create a wrapper for the TreeSelect component
- Implement CSF3 stories for TreeSelect
- Add unit tests using RTL

EnxDev committed:
chore(licences): add missing licences

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
fix(tag): resolve type errors

EnxDev committed:
refactor(imports): remove direct imports of antd components from src/components/index and create wrappers for missing components

EnxDev committed:
refactor(select): restore old structure for Select component and add index file for export

EnxDev committed:
refactor(upload): upgrade Upload component to AntD v5, update imports, add tests and story
- Replace all imports to be compatible with Ant Design v5
- Update the Upload component to align with new API changes
- Add unit tests to ensure correct functionality
- Create a Storybook story for visual testing

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
chore(license): add missing license for Upload.test

EnxDev commented on superset-frontend/src/components/Input/index.tsx:
```diff
@@ -17,10 +17,13 @@
 export const InputNumber = AntdInputNumber;
 
-export const { TextArea } = AntdInput;
+export const Input = Object.assign(Antd5Input, {
```
I did this to make the component more readable and understandable within our project. 
This way, the structure of the component is immediately clear to anyone who looks at it.

Should I remove it?

msyavuz replied:
I think yes, if we don't make any modifications to Input assigning what's already in it seems unneccessary. I missed this earlier.

geido replied:
I think this is unnecessary and we can safely remove it. It creates additional maintenance on the component if we have to upgrade to a newer version in the future.

EnxDev commented on superset-frontend/src/components/Layout/Layout.test.tsx:
```diff
@@ -0,0 +1,112 @@
+
+describe('Layout Component', () => {
+  const renderWithTheme = (ui: ReactElement) =>
+    render(<ThemeProvider theme={supersetTheme}>{ui}</ThemeProvider>);
```
Inside the `Layout.test.tsx` file, the Button component uses the `useTheme()` hook internally. For this reason, it is necessary to provide the ThemeProvider; otherwise, the test would fail.

geido replied:
@EnxDev we should not import from `@testing-library/react';` but from `'spec/helpers/testing-library'`

mistercrunch replied:
we should add a lint rule for this!

EnxDev committed:
refactor(Grid): add grid template to display multiple rows

EnxDev committed:
fix RTL DashboardBuilder.test.tsx

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
feat(Radio): add tests for Radio component

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
fix(Icons): change imports for Icons component

EnxDev committed:
fix components imports

EnxDev committed:
fix imports for select component

EnxDev committed:
replace imports for Tag component

EnxDev committed:
fix RTL tests

EnxDev committed:
fix RTL tests for Upload and Timer components

EnxDev committed:
remove custom icons from Tag component

EnxDev committed:
fix(Tag): fix RTL tests related to the close button on the Tag component

msyavuz commented on superset-frontend/src/components/Chart/DrillDetail/DrillDetailTableControls.test.tsx:
```diff
@@ -101,6 +101,6 @@ test('should remove the filters on close', () => {
   expect(screen.getByText('GB')).toBeInTheDocument();
 
-  userEvent.click(screen.getByRole('button', { name: 'close' }));
+  userEvent.click(screen.getByRole('img', { name: 'close' }));
```
Do we know why this change is necessary? I suspect this means close button is not accessible now.

EnxDev replied:
The AntD component references it as img.

msyavuz commented on superset-frontend/src/components/Layout/index.tsx:
```diff
@@ -17,5 +17,7 @@
+import type { LayoutProps, SiderProps } from 'antd-v5/es/layout';
 
+export type { LayoutProps, SiderProps };
 export default Layout;
```
We can change this to named exports as well

EnxDev replied:
Follow this https://github.com/apache/superset/pull/32705#discussion_r2020690125

msyavuz commented on superset-frontend/src/components/Radio/Radio.test.tsx:
```diff
@@ -0,0 +1,53 @@
+import '@testing-library/jest-dom';
+import Button from 'src/components/Button';
+
+describe('Button Component', () => {
+  test('renders button with text', () => {
+    render(
+      <ThemeProvider theme={supersetTheme}>
+        <Button>Click Me</Button>
+      </ThemeProvider>,
+    );
+    expect(screen.getByText('Click Me')).toBeInTheDocument();
+  });
+
+  test('calls onClick handler when clicked', () => {
+    const handleClick = jest.fn();
+    render(
+      <ThemeProvider theme={supersetTheme}>
+        <Button onClick={handleClick}>Click Me</Button>
+      </ThemeProvider>,
+    );
+    fireEvent.click(screen.getByText('Click Me'));
+    expect(handleClick).toHaveBeenCalledTimes(1);
+  });
+
+  test('renders a disabled button', () => {
+    render(
+      <ThemeProvider theme={supersetTheme}>
+        <Button disabled>Disabled</Button>
+      </ThemeProvider>,
+    );
+    expect(screen.getByRole('button', { name: 'Disabled' })).toBeDisabled();
+  });
+});
```
This test seems for a `Button` component but it's a `Radio` test?

msyavuz commented on superset-frontend/src/components/Select/AsyncSelect.stories.tsx:
```diff
@@ -18,12 +18,12 @@
   AsyncSelectRef,
   SelectOptionsTypePage,
-} from './types';
+} from 'src/components/Select/types';
+import { AsyncSelect } from '.';
```
Let's stick to one import either relative `./` or from `src/components`

EnxDev replied:
why not '.'; ?

msyavuz replied:
That's fine as well. But i think either use `.` or `src/components/Select/types`.

EnxDev replied:
Oh, you meant that both should be imported from '.'.

msyavuz commented on superset-frontend/src/components/Skeleton/index.tsx:
```diff
@@ -16,6 +16,6 @@
 
-export { default as TagsList } from './TagsList';
-export { default as Tag } from './Tag';
+export default Skeleton;
```
We can change this to named export as well

EnxDev replied:
@geido  and I were discussing modifying them in all components in a separate PR. @geido , what do you suggest we do?

Should I change the export types from default to named for all components in this PR, or should we open a separate one?

geido replied:
We will change these in a separate PR

msyavuz commented on superset-frontend/src/components/Tag/Tag.test.tsx:
```diff
@@ -37,7 +37,7 @@ test('should render shortname properly', () => {
   expect(container).toBeInTheDocument();
   expect(screen.getByTestId('tag')).toBeInTheDocument();
-  expect(screen.getByTestId('tag')).toHaveTextContent(mockedProps.name);
+  expect(screen.getByTestId('tag')).toHaveTextContent('example-tag');
```
Is this necessary? i think it's better to use mocked props for assertion instead of hard coded strings.

EnxDev replied:
I've re-added the prop, fixing the type error.

msyavuz commented on superset-frontend/src/components/Tag/utils.tsx:
```diff
@@ -37,9 +37,9 @@ const cachedSupersetGet = cacheWrapper(
 type SelectTagsValue = {
-  value: number | undefined;
+  value: number | string | undefined;
   label: string | undefined;
-  key: number | undefined;
+  key: number | string | undefined;
```
This seems like a functional change?

msyavuz commented on superset-frontend/src/components/Timer/Timer.test.tsx:
```diff
@@ -53,17 +58,17 @@ describe('Timer', () => {
     screen.rerender(<Timer {...mockProps} isRunning />);
     expect(screen.getByRole('timer')).toBe(node);
-    expect(node).toHaveTextContent(text);
+    expect(formatTime(node.textContent || '')).toBe(formatTime(text));
```
I think only the change in `Label` component might necessitate this but the styles look the same there so wondering why Timer needs to have different text?

EnxDev replied:
For simplicity, I can compare node.textContent with text. Comparing only node with text, as it was set before, is incorrect.

EnxDev replied:
The test was failing because there was a blank space, I have removed it from 'text'

msyavuz commented on superset-frontend/src/components/TreeSelect/index.tsx:
```diff
@@ -0,0 +1,23 @@
+import { TreeSelect } from 'antd-v5';
+import type { TreeSelectProps } from 'antd-v5/es/tree-select';
+
+export default TreeSelect;
```
Let's use named exports

EnxDev replied:
Follow this https://github.com/apache/superset/pull/32705#discussion_r2020690125

msyavuz commented on superset-frontend/src/components/Typography/index.tsx:
```diff
@@ -0,0 +1,21 @@
+ */
+import { Typography } from 'antd-v5';
+
+export default Typography;
```
Same here

EnxDev replied:
Follow this https://github.com/apache/superset/pull/32705#discussion_r2020690125

msyavuz commented on superset-frontend/src/components/Upload/index.tsx:
```diff
@@ -0,0 +1,23 @@
+import { Upload, type UploadFile } from 'antd-v5';
+import { type UploadChangeParam } from 'antd-v5/es/upload';
+
+export default Upload;
```
Same here

EnxDev replied:
Follow this https://github.com/apache/superset/pull/32705#discussion_r2020690125

msyavuz reviewed (commented):
Some feedback but mostly nits

michael-s-molina commented:
Thank you so much adding folders for each component! One suggestion though is to keep the previous import from `src/components` similar to what Ant Design and `@superset-ui/core` do. This folder will probably be moved to its own NPM package later.

```
import { AsyncSelect } from 'src/components'; <-- prefer this
import { AsyncSelect } from 'src/components/Select';
```

as later we'll probably end up with


```
import { AsyncSelect } from '@superset-ui/components';
```

EnxDev committed:
replace tests for Radio component

EnxDev committed:
fix type in Tag.test

EnxDev committed:
refactor(Timer test): Remove the blank space in the timer string in the test

EnxDev committed:
refactor(AsyncSelect story): Change the import type for the types

EnxDev committed:
fix type in Label component

EnxDev committed:
refactor(Upload, Layout): remove themeProvider from Upload.test and Layout.test

EnxDev committed:
refactor(input): change export for Input component

EnxDev committed:
refactor(Select, AsyncSelect): add export on src/components/index for Select and AsyncSelect components

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
remove double import for FormInstance

EnxDev committed:
fix import for AsyncSelect in SaveModal.test

geido committed:
Merge branch 'enxdev/refactor/antd5/create-wrappers' of https://github.com/apache/superset into enxdev/refactor/antd5/create-wrappers

geido commented:
Putting this on hold as I believe stories might need some thoughts

geido committed:
chore(Storybook): Enhance AutoComplete and Grid

EnxDev committed:
refactor(Upload): improve Upload story

EnxDev committed:
refactor(Layout): enhance the Layout story

EnxDev committed:
refactor(Skeleton): improve Skeleton story

EnxDev committed:
refactor(Tag): enhance the Tag story

EnxDev committed:
refactor(TagsList): optimize controls and cleanup story structure

EnxDev committed:
refactor(Tree): optimize controls for Tree story

EnxDev committed:
refactor(Typography): improve the Typography story

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/antd5/create-wrappers

EnxDev committed:
refactor(TreeSelect): improve story

EnxDev committed:
clean up

geido reviewed (approved):
Thanks!

michael-s-molina commented:
@geido @EnxDev It seems this [comment](https://github.com/apache/superset/pull/32705#issuecomment-2765931501) got lost and we merge it without taking it into consideration. Can we open a follow-up to address this?

mistercrunch commented:
Quick note that we can pile up some missed item in the `template_less` branch (the target for this PR). Trying to go high-velocity on the theming feature branch, and we have an opportunity to do deep testing and align anything we missed prior to merging `template_less` into `master`.

geido commented:
> import { AsyncSelect }

@michael-s-molina I am not sure what you are referring to specifically? I am looking into the codebase and it seems that currently we are doing this everywhere:

`import { AsyncSelect } from 'src/components';`

If you intended to apply this approach to all components, that is currently being developed in a separate PR. I have mentioned it somewhere else. It's under my radar currently.

Thank you!

michael-s-molina commented:
> @michael-s-molina I am not sure what you are referring to specifically?

@geido I'm talking about the following type of change that was made in this PR:

```
- import { Skeleton } from 'src/components';
+ import Skeleton from 'src/components/Skeleton';
```

Previously, it was importing from `src/components` as recommended, and now it's importing from `src/components/Skeleton`. There are many occurrences of this change.

EnxDev commented:
> > @michael-s-molina I am not sure what you are referring to specifically?
> 
> 
> 
> @geido I'm talking about the following type of change that was made in this PR:
> 
> 
> 
> ```
> 
> - import { Skeleton } from 'src/components';
> 
> + import Skeleton from 'src/components/Skeleton';
> 
> ```
> 
> 
> 
> Previously, it was importing from `src/components` as recommended, and now it's importing from `src/components/Skeleton`. There are many occurrences of this change.

I think at this point we can consider including it for all the components.

What do you think about it?

geido commented:
> > I think at this point we can consider including it for all the components.
> 
> As I mentioned [here](https://github.com/apache/superset/pull/32705#issuecomment-2765931501), these components will be moved later to an NPM package and consumers will probably import them like:
> 
> ```
> import { Skeleton } from '@superset-ui/components';
> ```
> 
> so it will be easier to migrate later if their current import is:
> 
> ```
> import { Skeleton } from 'src/components';
> ```
> 
> This also aligns with how we import Ant Design and `@superset/core` components.

@michael-s-molina it is in our plans to move all components inside ui-core as part of the theming effort and prior to that to change all of the imports to use `src/components. It is true that some got changed in the wrong direction in this PR but we were aware of it and accepted it as part of the feature branch. I am personally going to open a PR to fix all of these and all others in the codebase soon. Thanks for catching this!

michael-s-molina commented:
>  I am personally going to open a PR to fix all of these and all others in the codebase soon. Thanks for catching this!

No problem! Thank you all for the amazing work regarding Ant Design migration and theming!
